uui/components/pickers/__tests__/PickerInput.test.tsx

```diff
@@ -121,51 +161,921 @@ describe('PickerInput', () => {
+            expect(screen.queryAllByRole('button')).toHaveLength(3); // 2 tags and 1 clear button
+            const a1 = screen.getByText('A1');
+            const a1Clear = a1.nextElementSibling;
+            const a1plus = screen.getByText('A1+');
+            const a1plusClear = a1plus.nextElementSibling;
+            fireEvent.click(a1Clear as HTMLElement);
+            expect(screen.queryAllByRole('button')).toHaveLength(2); // 1 tag and 1 clear button
+            fireEvent.click(a1plusClear as HTMLElement);
+            expect(screen.queryAllByRole('button')).toHaveLength(0);
+        });
+        
+        it('should render names of items by getName', async () => {
+            const { dom } = await setupPickerInputForTest<TestItemType, number>({
+                value: [3, 4],
+                selectionMode: 'multi',
+                getName: ({ name }) => name,
+            });
+    
+            const selectedItemsNames1 = screen.queryAllByTestId(/uui-PickerToggler-item/)
+                .map((button) => button.textContent?.trim());
+            
+            expect(selectedItemsNames1).toEqual(['', '']);
+    
+            expect(dom.input?.getAttribute('placeholder')?.trim()).toEqual('Please select');
+       
+            await delayAct(100);
+    
+            const selectedItemsNames2 = screen.queryAllByTestId(/uui-PickerToggler-item/)
+                .map((button) => button.textContent?.trim());
+            
+            expect(selectedItemsNames2).toEqual(['Elementary+', 'Pre-Intermediate']);
+        });
+    
+        it('should render entity name with \'s\' in placeholder', async () => {
+            const { dom } = await setupPickerInputForTest({
+                value: undefined,
+                selectionMode: 'multi',
+                entityName: 'Language Level',
+            });
+            
+            expect(dom.input?.getAttribute('placeholder')?.trim()).toEqual('Please select Language Levels');
+        });
+    
+        it('should render plural entity name in placeholder', async () => {
+            const { dom } = await setupPickerInputForTest({
+                value: undefined,
+                selectionMode: 'multi',
+                entityName: 'Language Level',
+                entityPluralName: 'Multiple Language Levels',
+            });
+            
+            expect(dom.input?.getAttribute('placeholder')?.trim()).toEqual('Please select Multiple Language Levels');
+        });
+        
+        it('should pick single element with cascadeSelection = false', async () => {
+            const { mocks, dom } = await setupPickerInputForTest({
+                value: undefined,
+                getName: ({ name }) => name,
+                selectionMode: 'multi',
+                cascadeSelection: false,
+                dataSource: mockTreeLikeDataSourceAsync,
+            });
+    
+            fireEvent.click(dom.input as HTMLElement);
+            await delayAct(100);
```

Comment:
please avoid such kind of delays. use findBy* queries instead (https://testing-library.com/docs/guide-disappearance/)